tap_github/authenticator.py

```diff
@@ -381,3 +372,42 @@ def authenticate_request(
                 "For higher rate limits, please specify `auth_token` in config."
             )
         return request
+
+
+class MultiInstanceGitHubAuthenticator(GitHubTokenAuthenticator):
```

Comment:
If I understand correctly, the reason for this extension is to be able to create a single meltano task which is going to fetch data from multiple GH on prem instances? How many instances do we have? If it's just a8c and tumblr, wouldn't it be easier to simply create two independent tasks for each instance? That's what we usually do e.g. we have 4 hubspot instances and 4 pipelines. This also allows the system to be more resilient and transparent, because if one instance dies, the other pipelines work without interruption.

Reply:
Yes, it's so we can support GitHub Enterprise instances. We can definitely create an independent task for each instance, though in some cases, all three may write to the same stream/table. E.g both these will write to the `pr` table/stream -

```
          # GitHub.com 
          - api_url_base: https://api.github.com
            instance: github.com
            auth_token: ${TAP_GITHUB_EXAMPLE1_TOKEN}
            streams: ["pr"]
            org:
              - org1
          
          # GitHub Enterprise instance
          - api_url_base: https://github.example2.net/api
            instance: github.example2.net
            auth_token: ${TAP_GITHUB_EXAMPLE2_TOKEN}
            streams: ["pr"]
            org:
              - org2
```